src/game/PetHandler.cpp

```diff
@@ -662,7 +662,7 @@ void WorldSession::HandlePetSpellAutocastOpcode(WorldPacket& recvPacket)
     }
 
     // do not add not learned spells/ passive spells
-    if (!pet->HasSpell(spellid) || IsPassiveSpell(spellid))
+    if (!pet->HasSpell(spellid) || !IsAutocastable(spellid))
```

Comment:
This is basically the logic:

Execute if we don't have A, also execute if we don't have B.

Don't execute if we have both A and B.

You take a guess which logical operation fits which statement and then you tell me which makes the most logical sense and is easier to read. Just making a point since you call the readability of my logic horrible.